nbgrader/exchange/ngshare/exchange.py

```diff
@@ -18,10 +19,60 @@
 
 class Exchange(ABCExchange):
 
-    ngshare_url = 'http://172.17.0.3:11111'  # need to get IP address of container
-    username = os.environ['USER'] 
+    username = os.environ['JUPYTERHUB_USER']
```

Comment:
Fallback in case nbgrader is called outside of JupyterHub?

Reply:
Fixed